packages/plans-grid-next/src/components/test/header-price.tsx

```diff
@@ -73,11 +78,88 @@ describe( 'PlanFeatures2023GridHeaderPrice', () => {
 			},
 		} ) );
 
-		const { container } = render( <PlanFeatures2023GridHeaderPrice { ...defaultProps } /> );
+		const { container } = render( <HeaderPrice { ...defaultProps } /> );
 		const rawPrice = container.querySelector( '.plan-price' );
 		const discountedPrice = container.querySelector( '.plan-price.is-discounted' );
 
 		expect( rawPrice ).toHaveTextContent( '10' );
 		expect( discountedPrice ).toBeNull();
 	} );
+
+	test( 'should render empty for the enterprise plan', () => {
```

Comment:
Not sure what the value is of this test. It actually adds stronger binding to a specific plan. The idea/spec should be generic of the form "renders xyz for missing price /non-priced plans"

Reply:
if that's not the case, then it's a signal the logic needs updating, not fostered further through testing.

Reply:
@chriskmnds 

> Not sure what the value is of this test. 

It adds a test for the current behaviour related to `isWpcomEnterpriseGridPlan()`. Looking at the code paths, there should also be another test for `! isPricedPlan`. I'll add that.

> It actually adds stronger binding to a specific plan.

I'm not sure what this means.

> The idea/spec should be generic

I disagree. Specs should be as specific/targeted as possible, no?

> then it's a signal the logic needs updating, not fostered further through testing

Can you provide clarity around why you're not in favour of creating tests for the current logic?  Are you saying that not showing a price for the enterprise plan is a bug?

Reply:
I'm saying that "plan specific" functionality in the grid is a design-flaw e.g. conditioning on "premium" or "creator", etc. The logic should be abstracted. It's also only Enterprise plan that remains conditioned at places, and that needs to change (and it is possible to change). This test just adds to that situation.

A test around `sPricedPlan` plan should covered everything AFAIC